src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -501,7 +501,12 @@ protected override bool TryExec(Guid commandGroup, uint commandId)
                             var project = this.Workspace.CurrentSolution.GetProject(symbolListItem.ProjectId);
                             if (project != null)
                             {
-                                FindReferences(streamingPresenter, symbolListItem, project);
+                                // Note: we kick of FindReferencesAsync in a 'fire and forget' manner.
+                                // We don't want to block the UI thread while we compute the references,
+                                // and the references will be asynchronously added to the FindReferences
+                                // window as they are computed.  The user also knows something is happening
+                                // as the window, with the progress-banner will pop up immediately.
+                                var task = FindReferencesAsync(streamingPresenter, symbolListItem, project);
```

Comment:
:question: Does this need a `Task.Run` to ensure the operation really runs in the background?